### ISSUE TITLE
wrong result for references

### ISSUE THREAD

**Issue by qmmp123** [microsoft/pylance-release#4285]
**Describe the bug**
pyright returns wrong result when try to get references

**To Reproduce**
Create file foo.py with content
```python
def function():
  pass
```
Create file bar.py with content
```python
def function():
  pass
```
Create file baz.py with content
```python
from foo import function
function()
```
In foo.py try to get references of `function` this will return also `function` from bar.py

**Expected behavior**
I think pyright should not return references from bar.py

**command-line**
pyright v1.1.303
neovim v0.9.0

**Comment by Azureblade3808** [comment]
How to repro
--

```python
# a.py
from .b import *
from .c import *
```

```python
# b.py
import builtins
s = "b"
```

```python
# c.py
import builtins
s = "c"
```

Use `Find all references` on `s` in either "b.py" or "c.py", and both `s`s will be listed.

![image](https://user-images.githubusercontent.com/15389744/234766024-91550bab-52f7-4c98-be0d-bec377568405.png)

**Comment by Azureblade3808** [comment]
- "a.py" must import `s` from both "b.py" and "c.py" while star imports are not required.
- "b.py" and "c.py" must both import a common module while it is not required to be `builtins`.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.5.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023521-10-May-2023-PreRelease)

**Comment by last-partizan** [comment]
Is this fix included in pyright 1.1.307?

**Comment by debonte** [comment]
> Is this fix included in pyright 1.1.307?

No, it will be in Pyright's next release.
